src/main/kotlin/it/pagopa/ecommerce/scheduler/queues/TransactionActivatedEventsConsumer.kt

```diff
@@ -101,49 +100,61 @@ class TransactionActivatedEventsConsumer(
                 transaction.transactionId.value.toString(),
-                transaction.rptId.value,
-                paymentToken,
                 TransactionExpiredData(transaction.status)
             )
         ).then(
             transactionsViewRepository.save(
                 Transaction(
                     transaction.transactionId.value.toString(),
-                    paymentToken,
-                    transaction.rptId.value,
-                    transaction.description.value,
-                    transaction.amount.value,
+                    transaction.paymentNotices.map { notice ->
+                                                   PaymentNotice(
+                                                       notice.paymentToken.value,
+                                                       notice.rptId.value,
+                                                       notice.transactionDescription.value,
+                                                       notice.transactionAmount.value,
+                                                       notice.paymentContextCode.value
+                                                   )
+                    },
+                    transaction.paymentNotices.sumOf { it.transactionAmount.value },
                     transaction.email.value,
-                    TransactionStatusDto.EXPIRED
+                    TransactionStatusDto.EXPIRED,
+                    transaction.clientId,
+                    transaction.creationDate.toString()
                 )
             ).then()
         )
     }
 
-    private fun updateTransactionToRefunded(transaction: BaseTransaction, paymentToken: String): Mono<Void> {
+    private fun updateTransactionToRefunded(transaction: BaseTransaction): Mono<Void> {
 
         return transactionsRefundedEventStoreRepository.save(
             TransactionRefundedEvent(
                 transaction.transactionId.value.toString(),
-                transaction.rptId.value,
-                paymentToken,
                 TransactionRefundedData(transaction.status)
             )
         ).then(
             transactionsViewRepository.save(
                 Transaction(
                     transaction.transactionId.value.toString(),
-                    paymentToken,
-                    transaction.rptId.value,
-                    transaction.description.value,
-                    transaction.amount.value,
+                    transaction.paymentNotices.map { notice ->
+                        PaymentNotice(
+                            notice.paymentToken.value,
+                            notice.rptId.value,
+                            notice.transactionDescription.value,
+                            notice.transactionAmount.value,
+                            notice.paymentContextCode.value
+                        )
+                    },
+                    transaction.paymentNotices.sumOf { it.transactionAmount.value },
```

Comment:
move this code to a method to avoid duplicated code

Reply:
Fixed in 0465d23